pinot-segment-local/src/main/java/org/apache/pinot/segment/local/utils/HashUtils.java

```diff
@@ -44,6 +77,8 @@ public static Object hashPrimaryKey(PrimaryKey primaryKey, HashFunction hashFunc
         return new ByteArray(HashUtils.hashMD5(primaryKey.asBytes()));
       case MURMUR3:
         return new ByteArray(HashUtils.hashMurmur3(primaryKey.asBytes()));
+      case UUID_V4:
+        return new ByteArray(HashUtils.hashUUIDv4(primaryKey.asBytes()));
```

Comment:
self-review: for cases with multiple columns as primary-keys,`PrimaryKey#asBytes` prepends each string with 4-bytes to denote the length of the following string. So this won't work directly. Maybe consider using `PrimaryKey#getValues`?